src/com/linkedin/parseq/internal/SerialExecutor.java

```diff
@@ -117,7 +122,12 @@ public void run() {
       // Entering state:
       // - _queue.size() > 0
       // - _pendingCount.get() > 0
+
+      final ExecutionMonitorState executionState = _executionMonitor != null ? _executionMonitor.getLocalMonitorState() : null;
```

Comment:
Yeah that was my first thought too but in order to achieve it I would have to create an abstract class or interface with implementation which adds a bit of complexity. Not sure if this is worth an effort vs few null checks that are completely internal to SerialExecutor - it does not leak out of this class.